error.go

```diff
@@ -0,0 +1,35 @@
+package main
+
+import "fmt"
+
+// We define a custom error type so that we can provide friendlier error messages
+type fetchError struct {
```

Comment:
I was thinking about that, but I wouldn't want this type to be used outside of this library, so I decided to default to lowercase. Do you have a different take?